### PR TITLE
feat: Add Proxy Support

### PR DESCRIPTION
**Pull request by riley-kohler**
# Pull Request

## Proposed Changes

This PR adds proxy support to ICF by configuring fetch/undici with a global dispatcher the uses proxy envs and older request libraries with a global agent that uses the envs as well. 

This solution works but has a couple of remaining issues. 
- It required changing the webhook relay to typescript so that it can import the same proxy code but exposed some existing type issues in that file which remain unfixed. 
- Due to ESM and CommonJS compatibility issues, this change required switching to tsx instead of ts-node which no longer checks types at runtime. This can potentially be handled through explicitly running type checking as part of a lint step or when building. 
- The proxy support requires some transitive dependencies which are not available when running as a client (in a browser) which means that it cannot be explicitly added to "src/bot/octokit.ts" since that code is called from both the client and server contexts. In practice, this hasn't been an issue as the nextauth import always seems to happen beforehand but this may cause issues in the future.

I wanted to put up this PR to get the "working" proxy support out there but am open to any suggestions or changes needed before merging.

### Author/Contributor

- [ ] If documentation is needed for this change, has that been included in this pull request
- [ ] run `npm run lint` and fix any linting issues that have been introduced
- [ ] run `npm run test` and run tests

### Reviewer

- [ ] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, `maintenance`, or `breaking`

### CODE REVIEW

**Comment by ajhenry** [comment]
Hey all, gonna be taking care of this in the next day or so! I'm going to remove `scripts` from the docker build step since the forwarder shouldn't be needed there. 

@sutterj will be taking point on this 👑

**Comment by sutterj** [comment]
I'm looking into converting the project/tests to ESM to see if I can't solve this issue more elegantly, but I was able to get it to work by making a second version of the proxy as mjs and converting the webhooks relay back to mjs.  Not the "ideal" solution, but it solves the types issues and allows us to go back to ts-node.

**Comment by sutterj** [comment]
@riley-kohler Can you pull the latest here and test it on your network?

**Review by ajhenry — APPROVED** [review]
This looks good from my end, I'll wait for confirmation to ensure this works properly.

As for duplicating the proxy setup — that's because scripts is excluded from our build step in tsconfig and since its a relatively small change, I'll cool with keeping it as is

**Comment by riley-kohler** [comment]
Just tested on our network and everything is still working

File: src/utils/proxy.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import http from 'http'
+import https from 'https'
+
+// set unidci global dispatcher to a proxy agent based on env variables for fetch calls
+const envHttpProxyAgent = new EnvHttpProxyAgent()
+setGlobalDispatcher(envHttpProxyAgent)
+
+// set global agent for older libraries that use http and https calls
+const proxyAgent = new ProxyAgent()
+http.globalAgent = proxyAgent
```
Comment (zkoppert) [inline src/utils/proxy.ts:12]: Do we need to allow http? May be a security risk.

Comment (ajhenry) [reply src/utils/proxy.ts:12]: Afaik both http and https will need to be set for proxy requests to work properly

Comment (zkoppert) [reply src/utils/proxy.ts:12]: understood, Thanks!

**Review by zkoppert — APPROVED** [review]

**Pull request merged**